### ISSUE TITLE
Not an Issue -- just a  message 

### ISSUE THREAD

**Issue by garyStofer** [thetazzbot/FreeExpression#1]
I see you just checked in your code. I forked it and will add the modified LCD stuff to it from my version of Arlet's Freecut. It's using a standard 4x40 lcd display instead of what came with the machine. 

What are you using to feed hpgl code to the machine ? 

Gary

**Comment by markawilliamson** [comment]
It uses serial, xon/xoff.  The 4x40 lcd does not come on the Cricut Expressions.  Beware, that this code is only for Cricut Expression.  Would need some effort to make it work on the mini/personal.  Let me know how it goes.

**Comment by garyStofer** [comment]
No no .. I have a cake machine and have already ported Arlet's code to work on the PIC 1281 -- . The cake machine has the same electrics as the Expression machine. I think It's actually the exact same machine except for color and the cutter used as the larger of the Expression ones. 

I added code for a standard 4x40 LCD that i added to the machine by cutting the original graphic LCD out and hocking up my LCD up instead. 

I was wondering what you are using to  get HPGL output from a drawing program -- are you using pstoedit http://www.pstoedit.net/pstoedit  or something similar or are you setting up a HP-Plotter and then use Print to File to get the output  ??  When I do that on Windows I get a lot of stuff in the HPGL file that I don't recognize.  Are you running the machine from a linux or windows box?  Thanks  Gary

**Comment by markawilliamson** [comment]
ah, right

why did you take out the lcd?

Inkscape version .91pre3 and above has the hpgl plugin, you can cut
directly from inkscape :)

have to download the dev version though.

I'm running from windows.

http://www.thetazzbot.com/2015/02/16/cutting-my-first-design-from-inkscape-to-my-craft-cutter/

**Comment by garyStofer** [comment]
Ok --  Sweet !  I have Inkscape running on the laptop. I didn't know they have an HPGL output. That's excellent.  

I removed the LCD because I could not figure out how the LCD worked and I already had flashed the PIC, so I could not hock up the logic analyzer and see what was going on on the pins.

**Comment by garyStofer** [comment]
What's the correct DPI setting for the HPGL  export ? I got the code modified to work with  my display and the machine is moving as expected, but the scale is about 3 times too big.  I figure I need a different DPI setting from the default. Also is the pressure setting and speed setting controlled from the HPGL input or is it only taken from the settings given by the keyboard? 

Regarding lcd -- Mine is actually a SPI version -- Less wire intensive than the standard ones.

**Comment by markawilliamson** [comment]
pressure and speed are not controlled by hpgl yet.  use the arrow keys

the dpi should be 400 for each x and y

glad you got it working!

**Comment by markawilliamson** [comment]
I some how managed to fry the ADC on my board so my wheels dont work.

Can you do some testing on that for me? They should work as is right now,
so if you move either speed or pressure wheel the display should say
something...

**Comment by garyStofer** [comment]
Ok, thanks -- Yes , 400 will make it 1:1 

Just checked the potentiometer wheels, but there is no indication that it does anything. I see that the +/- buttons change the display.

I can look at that with the debugger next week. I brought JTAG out to the place where the cartridge used to be.  

Gary

**Comment by markawilliamson** [comment]
hahaha

thats exactly what I did.  Love that they made it easy.

Oh, pick up some cartridges from craigs list.  They have a nice atmel chip
with an eeprom chip in there ;)

I had some devious plans for that but decided specifically not to use the
cartridges in this hack because that is what Provo sued SCAL and Make the
cut over...  Apparently they have a patent on that.

but the cartridge boards could be adapted to other uses.  i have some 3d
printer projects in the works that i might use them.

**Comment by markawilliamson** [comment]
sorry, not eeprom, flash

**Comment by garyStofer** [comment]
Hey Mark,

I was looking at the ADC issue today --  Are you sure you have the 5V to the left hand potentiometers still present? It's Pin1 on the ribbon cable going to the two LH pots. In the original electronic that voltage was witched by Q2 via R28 from pin PF4, (ADC4, TCK).  

Same switched power is also used for the LCD back light and the RH dial.  In my modification I rerouted TCK (JTAG signal) to the front connector and strapped the cut side of R28 to ground to permanently enable power to the pots.

Also, The right hand dial is not a potentiometer if the dial freely rotates past 360 degrees ( has no stop) It's actually a quadrature encoder. There are two jacks on the PCB. If a 4 wire cable is connected to J3 you have the quad, else if a 3 wire cable is connectd to J2 you have a pot.  Quadrature encoder pins are connected to PE4 (int4, pin 6) and PF0 (ADC0, pin 61).

I have most of this sketched out by hand, along with my Jtag cable and SPI lcd modifications.  If you want I can put it into Eagle Cad or scan a copy of my hand notes. 

Will test the modified ADC code shortly, but I'm not sure If using the dials to set pressure and speed is preferred over using buttons or better yet input from HPGL. 

Cheers 

Gary

**Comment by markawilliamson** [comment]
Yeah the right side is not a pot, it is an encoder.  that part works.

the part that isnt working on mine are the left dials.

i have a another one i bought non-working the motherboard is dead.  thought
i could revive it but no go.

im going to dig it out and mess with it this weekend because i want to cut
a template.

**Comment by garyStofer** [comment]
Not much to the motherboard. If all else fails you can replace the MCU fairly easily, no difficult to solder parts on it. 

I didn't see where you had the q-encoder code -- didn't look for it either, but figured that since it had the potentiometer code in dial.c it was mistakenly using the wrong type. I added a #ifdef .

Make sure you have the +5V from Q2 on the pots as  I mentioned below.   You had some comments in the code about resistors that burned off. What where they. 

I made some changes to dial.c so that they set the speed and pressure in a range of 2-6 and then the +/- buttons can adjust the values further up or down. I noticed that speed 10 is too fast for the at least the Y motor to move properly, so maybe limit that to 0-9 

If you want, I can send you a pull request when I'm done with it -- Or I can send you the modified files in a zip.

I noticed that there is an InkCut plug-in. Have not tried it but looks awesome ( is missing some files under windows). I'm going to run the cutter from Linux anyways in the end. 

Cheers
     Gary

**Comment by garyStofer** [comment]
Forgot to ask you how you modified the board to get at the JTAG pins and still have the LCD working?  Dindn't they use some of the JTAG signals on that LCD interface ?

Gary

**Comment by markawilliamson** [comment]
The inkcut plugin is what ive been using.

**Comment by garyStofer** [comment]
I give up on the InkCut extension to Inkscape. I can't get it to run on a Windows nor a Linux system.  On Windows it complains about some pygtk module missing and aborts. Under Linux it crashes when I try to open the properties to configure the com port.  I see they have a bunch of issues ( bugs) registered on SourceForge and nothing is being addressed.

I guess it works well enough with the Inkscape built in Export to plotter that InkCut is not really needed anyway.  Or Maybe this is my clue to finally learn Python ?? 

Found that the pressure settings where not quite scaled right, so that pressure settings 8,9 and 10 created the same actual pressure on the cutter.  Also at the maximum pressure setting the solenoid is getting hot and the actual pressure falls because of it.  I measured a max pressure of 340 grams when cold which would decay to about 260 grams when the cuter was down for a minute or two.  Not good for consistency. I wonder how much pressure is needed ever. I will experiment with cutting some Mylar next. 

Gary

**Comment by garyStofer** [comment]
Hey !

Did you ever get Inkcut to run on the older V0.48 Inkscape running under Linux. 

I spent two full days now and I can't get this thing to not crash or give me some other errors. After finding one missing component after an other I got it as far as putting up the gui with the preview, but it immediately crashes when I try to open the preferences to set the comport. Other people have had similar problems.

I was hoping to get it running for the duplicate and position feature it has.  I need to space the cutting area a bit away from the machine edge because my 1 mm gasket material can't go under the feed rollers.  

I'm thinking now to add an offset feature to the machine itself, sort of like a "touch off" feature on a CNC so that the 0,0 can be positioned on the exact spot of the to be cut material. 

Gary
